lib/evse_security/evse_security.cpp

```diff
@@ -462,6 +463,35 @@ EvseSecurity::get_installed_certificates(const std::vector<CertificateType>& cer
     return result;
 }
 
+int EvseSecurity::get_count_of_installed_certificates(const std::vector<CertificateType>& certificate_types) {
+    int count = 0;
+
+    std::set<fs::path> directories;
+    const auto ca_certificate_types = get_ca_certificate_types(certificate_types);
+
+    // Collect unique directories
+    for (const auto& ca_certificate_type : ca_certificate_types) {
+        directories.emplace(this->ca_bundle_path_map.at(ca_certificate_type));
+    }
+
+    for (const auto& unique_dir : directories) {
+        X509CertificateBundle ca_bundle(unique_dir, EncodingFormat::PEM);
+        count += ca_bundle.get_certificate_count();
+    }
+
+    // V2G Chain
+    if (std::find(certificate_types.begin(), certificate_types.end(), CertificateType::V2GCertificateChain) !=
+        certificate_types.end()) {
+        auto leaf_dir = this->directories.secc_leaf_cert_directory;
+
+        // Load all from chain, including expired/unused
+        X509CertificateBundle leaf_bundle(leaf_dir, EncodingFormat::PEM);
+        count += leaf_bundle.get_certificate_count();
+    }
+
+    return count;
+}
+
```

Comment:
Question: Is it possible to count a certificate twice here? I know in some cases (e.g. sub-CAs) the same certificate can show up in multiple bundles.

Reply:
Yes, this does not take into account the unique certificates. For example, we can have the SECC_Leaf part of a file (SECC_LEAF.pem) and also a chain (CPO_CERT_CHAIN.pem), case in which it will be counted twice. The specification, here is not very clear. Do we want to count actual entries or only unique certificates?

Reply:
Feedback from Piet: Likely we want to count duplicates twice here - so this version should be fine.

Reply:
Feedback from Piet: This implementation should be fine as it is - the spec is a bit vague on this, but duplicates likely should be considered as multiple certificates.